### PR TITLE
Add BATTALION: Legacy

### LINKED ISSUE #2364: [Egg Request]: BATTALION: Legacy
### Does this expand an already existing service

No

### Link to a game or software

https://store.steampowered.com/app/489940/BATTALION_Legacy

### Links for the download

https://steamdb.info/app/805140

### Links for the install docs

https://steamcommunity.com/app/489940/discussions/0/3432326055078758588/

### PR DESCRIPTION
**Pull request by diedyesterdaywashere**
# Description

closes #2364 

shout-out to @gOOvER, he helped me with the egg a lot!

## Checklist for all submissions

* [X] Have you followed the guidelines in our [Contributing document](https://github.com/parkervcp/eggs/blob/master/CONTRIBUTING.md)?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [X] Have you tested and reviewed your changes with confidence that everything works?
* [X] Did you branch your changes and PR from that branch and not from your master branch?
  * If not, why?:

## New egg Submissions

1. [X] Does your submission pass tests (server is connectable)?
2. [ ] Does your egg use a custom docker image?
    * [ ] Have you tried to use a generic image?
    * [ ] Did you PR the necessary changes to make it work?
3. [X] Have you added the egg to the main README.md and any other README files in subdirectories of the egg (e.g /game_eggs) according to the alphabetical order?
4. [X] Have you added a unique README.md for the egg you are adding according to the alphabetical order?
5. [X] You verify that the start command applied does not use a shell script
    * [ ] If some script is needed then it is part of a current yolk or a PR to add one
6. [X] The egg was exported from the panel

### CODE REVIEW

**Review by gOOvER — APPROVED** [review]
egg works fine. Server is shown in serverlist and you can connect to it

File: game_eggs/steamcmd_servers/battalion_legacy/egg-b-a-t-t-a-l-i-o-n--legacy.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    },
+    "variables": [
+        {
+            "name": "App ID",
+            "description": "",
+            "env_variable": "SRCDS_APPID",
+            "default_value": "805140",
+            "user_viewable": true,
+            "user_editable": true,
+            "rules": "required|string|max:20",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/battalion_legacy/egg-b-a-t-t-a-l-i-o-n--legacy.json:40]: Makz this use the rule `required|string|in:805140`

File: game_eggs/steamcmd_servers/battalion_legacy/egg-b-a-t-t-a-l-i-o-n--legacy.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            "field_type": "text"
+        },
+        {
+            "name": "Auto Update",
```
Comment (QuintenQVD0) [inline game_eggs/steamcmd_servers/battalion_legacy/egg-b-a-t-t-a-l-i-o-n--legacy.json:44]: Add a mentione here that it only happens on restart.

**Review by QuintenQVD0 — CHANGES_REQUESTED** [review]
Just those 2, everything else is fine

**Comment by diedyesterdaywashere** [comment]
@QuintenQVD0 here you go, added your changes

**Review by QuintenQVD0 — APPROVED** [review]

**Comment by QuintenQVD0** [comment]
LGTM! Thank you.

**Pull request merged**